### PR TITLE
Add support for sqlx NamedStmt

### PR DESCRIPTION
**Pull request by Thiht**
Additionally to Stmt and Rows, sqlx exposes [NamedStmt](https://pkg.go.dev/github.com/jmoiron/sqlx#NamedStmt)
As for Stmt, they should also be closed to avoid pool exhaustion and potential leaks

### CODE REVIEW

**Review by ryanrolds — DISMISSED** [review]
Thank you. 

I intend to merge this after #25 is merged.

**Comment by ryanrolds** [comment]
There is some conflicts. Once those are resolved I will merge.

**Comment by ryanrolds** [comment]
I've enabled CI on PRs in `main.` Don't be surprised if you get lint errors or failing tests. If need help resolving CI failures, let me know.

**Comment by Thiht** [comment]
> There is some conflicts. Once those are resolved I will merge.

@ryanrolds  Updated!

**Comment by Thiht** [comment]
Sorry, just noticed didn't stage my changes 😅

Updated for real this time.

Lint/test/build OK in local, I don't think I can trigger the CI myself

**Comment by ryanrolds** [comment]
Sorry. I wasn't aware that GH wouldn't run CI automatically on this PR.

**Review by ryanrolds — APPROVED** [review]

**Comment by Thiht** [comment]
No problem, thanks for the merge!

**Pull request merged**
